sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SchemaMergeUtils.scala

```diff
@@ -80,7 +80,7 @@ object SchemaMergeUtils extends Logging {
             var mergedSchema = schemas.head
             schemas.tail.foreach { schema =>
               try {
-                mergedSchema = mergedSchema.merge(schema)
+                mergedSchema = mergedSchema.merge(schema, sparkSession.sessionState.conf.resolver)
```

Comment:
Trying to pull the resolver out of session state breaks a lot of the tests. I can revert most of these to not use the resolver and maintain the same behavior for now, but wanted to at least bring up the discussion if it makes sense to do resolver-aware schema merging here. Also, is there some other way you can/supposed to get a resolver? Or is it just a weird test thing?